ddsrouter_core/src/cpp/dynamic/DiscoveryDatabase.cpp

```diff
@@ -85,11 +105,16 @@ bool DiscoveryDatabase::add_endpoint(
 
         logInfo(DDSROUTER_DISCOVERY_DATABASE, "Inserting a new discovered Endpoint " << new_endpoint << ".");
 
+        for (auto discovered_endpoint_callback : discovered_endpoint_callbacks_)
+        {
+            discovered_endpoint_callback(new_endpoint);
+        }
+
         return true;
     }
 }
 
-bool DiscoveryDatabase::update_endpoint(
+bool DiscoveryDatabase::update_endpoint_(
```

Comment:
Shouldn't this method call the callbacks as well?

Reply:
The idea was to have a different vector of callbacks for each of the possible actions that modify the database: add, update and erase. I have added the other two vectors although for now they are empty and serve no purpose.

Reply:
Add TODO in header so the future developer knows a method is missing to register those callbacks.